lib/logstash/inputs/http.rb

```diff
@@ -138,16 +144,20 @@ def run(queue)
         REJECTED_HEADERS.each {|k| req.delete(k) }
         req = lowercase_keys(req)
         body = req.delete("rack.input")
-        @codec_lock.lock
+        local_codecs = @write_slots.poll()
+        if !local_codecs # No free write slot
+          next [429, {}, ['Write slots full']]
```

Comment:
Any reason why not returning response headers?

Not sure if `write slots full` is a meaningful error message for users?